javascript-custom-methods/calculator.js

```diff
@@ -0,0 +1,41 @@
+/* exported calculator */
+var calculator = {
+  add: function (x, y) {
+    return x + y;
+  },
+
+  subtract: function (x, y) {
+    return x - y;
+  },
+
+  multiply: function (x, y) {
+    return x * y;
+  },
+
+  divide: function (x, y) {
+    return x / y;
+  },
+
+  square: function (x) {
+    return x * x;
+  },
+
+  sumAll: function (numbers) {
+    var sum = 0;
+
+    for (var i = 0; i < numbers.length; i++) {
+      sum += numbers[i];
+    }
+    return sum;
+  },
+
+  getAverage: function (numbers) {
+    var avg = 0;
+    var indexLength = numbers.length;
+
+    for (var j = 0; j < numbers.length; j++) {
+      avg += numbers[j] / indexLength;
+    }
+    return avg;
```

Comment:
You should add all the numbers first, then divide the total by the length of the array after the loop ends